packages/experiments/src/components/Button/MenuButton/MenuButton.state.ts

```diff
@@ -1,68 +1,56 @@
-import * as React from 'react';
-import { BaseState } from '../../../utilities/BaseState';
-import { IMenuButtonProps, IMenuButtonViewProps } from './MenuButton.types';
-
-export type IMenuButtonState = Pick<
-  IMenuButtonViewProps,
-  'expanded' | 'onClick' | 'onKeyDown' | 'buttonRef' | 'onMenuDismiss' | 'menuTarget'
->;
-
-export class MenuButtonState extends BaseState<IMenuButtonProps, IMenuButtonViewProps, IMenuButtonState> {
-  private _buttonRef = React.createRef<HTMLButtonElement>();
-
-  constructor(props: MenuButtonState['props']) {
-    super(props, {
-      controlledProps: ['expanded']
-    });
-
-    this.state = {
-      expanded: !!props.defaultExpanded,
-      onClick: this._onClick,
-      onKeyDown: this._onKeyDown,
-      buttonRef: this._buttonRef,
-      onMenuDismiss: this._onMenuDismiss,
-      menuTarget: undefined
-    };
-  }
-
-  public focus(): void {
-    if (this._buttonRef.current) {
-      this._buttonRef.current.focus();
-    }
-  }
-
-  private _onMenuDismiss = () => {
-    this.setState({
-      expanded: false
-    });
-  };
-
-  private _onClick = (ev: React.MouseEvent<HTMLElement>) => {
-    const { disabled, onClick } = this.props;
-
-    if (!disabled) {
-      if (onClick) {
-        onClick(ev);
-
-        if (ev.defaultPrevented) {
-          return;
+import { useCallback, useState } from 'react';
+import { useControlledState } from '../../../Foundation';
+import { KeyCodes } from '../../../Utilities';
+import { IMenuButtonComponent, IMenuButtonViewProps } from './MenuButton.types';
+
+export const useMenuButtonState: IMenuButtonComponent['state'] = props => {
+  const [menuTarget, setMenuTarget] = useState<HTMLElement | undefined>(undefined);
+  const [expanded, setExpanded] = useControlledState(props, 'expanded', {
+    defaultPropName: 'defaultExpanded',
+    defaultPropValue: false
+  });
+
+  const _onMenuDismiss = useCallback(() => {
+    setExpanded(false);
+  }, []);
```

Comment:
General note due to some questions (and me forgetting temporarily): `useCallback` is used here to memoize the callback and prevent its value from changing on every render. [React documentation states that setState functions are 'stable'](https://reactjs.org/docs/hooks-faq.html#what-can-i-do-if-my-effect-dependencies-change-too-often) and therefore do not need to appear in deps array. The effect of an empty array is that only one version of the callback will ever be memoized.